platform/osx/export/export.cpp

```diff
@@ -574,34 +574,34 @@ Error EditorExportPlatformOSX::export_project(const Ref<EditorExportPreset> &p_p
 
 	String pkg_name_safe = OS::get_singleton()->get_safe_dir_name(pkg_name);
 
-	Error err = OK;
-	String tmp_app_path_name = "";
-
-	DirAccess *tmp_app_path = nullptr;
 	String export_format = use_dmg() && p_path.ends_with("dmg") ? "dmg" : "zip";
 
 	// Create our application bundle.
-	tmp_app_path_name = EditorSettings::get_singleton()->get_cache_dir().plus_file(pkg_name + ".app");
+	String tmp_app_dir_name = pkg_name + ".app";
```

Comment:
Maybe we should check that "pkg_name" does not have anything nasty, like "../../.." and won't delete anything outside cache.

Reply:
Should we use `pkg_name_safe` for that which we have a few lines above?

Reply:
Yes, probably. It should have ".." removed.

Reply:
Done. BTW the Mono-specific stuff might still need fixes for ARM64 support as currently it hardcodes `.64` but we'd need to disambiguate the arch.

Reply:
With exception to 3 `dylib`s, Mono related files should be platform independent. And for `dylib`s it's probably better to ship them as universal binaries. In this case all existing export code should work as is (`.64` could be renamed to `.universal` or something like this).